src/conversion.jl

```diff
@@ -46,6 +46,33 @@ Returns 1. (Avoids effort when unnecessary.)
 """
 convfact(s::Units{S}, t::Units{S}) where {S} = 1
 
+"""
+    convfact(T::Type, s::Units, t::Units)
+Returns the appropriate conversion factor from unit `t` to unit `s` for the number type `T`.
+"""
+@generated function convfact(::Type{T}, s::Units, t::Units) where T
+    cf = convfact(s(), t())
+    if cf isa AbstractFloat
+        F = floattype(T)
+        # Since conversion factors only have Float64 precision,
+        # there is no point in converting to BigFloat
+        convert(F == BigFloat ? Float64 : F, cf)
```

Comment:
Perhaps using `<:` instead of `==` is safer to prevent possible `BligFloat` type changes.
```suggestion
        convert(F <: BigFloat ? Float64 : F, cf)
```

Reply:
I think it is unlikely that `BigFloat` will be changed to a non-concrete type, but we can use `<:` if you prefer it.

Reply:
I think you're right. Perhaps just change`==` to `===` which is more common to compare types.

Reply:
I don’t really like using `===` on types, because the same type can have different representations (although it shouldn’t be a problem in this case because `BigFloat` is a concrete type):
```
julia> (Rational{T} where T) == (Rational{S} where S)
true

julia> (Rational{T} where T) === (Rational{S} where S)
false
```
I also didn’t find an instance of using `===` on types in this package, but there is one for `==`: https://github.com/PainterQubits/Unitful.jl/blob/f375caf8f7c60f0dc2e284a97fc902365f55ab76/src/conversion.jl#L67

Reply:
I see, that makes perfect sense.